h/h_api/model/json_api.py

```diff
@@ -0,0 +1,34 @@
+"""Generic model objects representing objects in JSON API style."""
+
+from h.h_api.model.base import Model
+
+
+class JSONAPIError(Model):
+    """A JSON API error wrapper."""
+
+    @classmethod
+    def create(cls, error_bodies):
+        return cls({"errors": [cls.extract_raw(body) for body in error_bodies]})
+
+
+class JSONAPIErrorBody(Model):
+    """A JSON API error body."""
+
+    @classmethod
+    def create(
+        cls, exception, title=None, detail=None, pointer=None, status=None, meta=None
+    ):
+        return cls(
+            cls.dict_from_populated(
+                code=exception.__class__.__name__,
+                title=title or exception.args[0],
+                detail=detail,
+                meta=meta,
+                status=str(int(status)) if status is not None else None,
+                source={"pointer": pointer} if pointer else None,
+            )
+        )
+
+    @property
+    def detail(self):
+        return self.raw.get("detail", None)
```

Comment:
So this is the anti-POJO method. Seems nicer indeed

Reply:
Yeah. It seems a bit nonsensy for a lot of properties like this one where the key and the property name happen to be 1:1, but for things that reach into the data structure more deeply or return puffed up objects, it makes more sense.